ts/modules/marker-clusters.ts

```diff
@@ -0,0 +1,1948 @@
+         * @optionparent plotOptions.scatter.marker.cluster.layoutAlgorithm.type
+         */
+        /* eslint-enable max-len */
+        /**
+         * When `type` is set to the `grid`,
+         * `gridSize` is a size of a grid square element either as a number
+         * defining pixels, or a percentage defining a percentage
+         * of the plot area width.
+         *
+         * @type    {number|string}
+         */
+        gridSize: 50,
+        /* eslint-disable max-len */
+        /**
+         * When `type` is set to `kmeans`,
+         * `iterations` are the number of iterations that this algorithm will be
+         * repeated to find clusters positions.
+         *
+         * @type    {number}
+         * @optionparent plotOptions.scatter.marker.cluster.layoutAlgorithm.iterations
+         */
+        /* eslint-enable max-len */
+        /**
+         * When `type` is set to `kmeans`,
+         * `distance` is a maximum distance between point and cluster center
+         * so that this point will be inside the cluster. The distance
+         * is either a number defining pixels or a percentage
+         * defining a percentage of the plot area width.
+         *
+         * @type    {number|string}
+         */
+        distance: 40,
+        /**
+         * When `type` is set to `undefined` and there are more visible points
+         * than the kmeansThreshold the `grid` algorithm is used to find
+         * clusters, otherwise `kmeans`. It ensures good performance on
+         * large datasets and better clusters arrangement after the zoom.
+         *
+         * @type    {number}
+         */
+        kmeansThreshold: 100
+    },
+    style: {
+        /**
+         * A predefined shape or symbol for the cluster marker. Other
+         * possible values are "circle", "square", "diamond", "triangle"
+         * and "triangle-down".
+         *
+         * Additionally, the URL to a graphic can be given on this form:
+         * "url(graphic.png)". Note that for the image to be applied to
+         * exported charts, its URL needs to be accessible by the export
+         * server.
+         *
+         * Custom callbacks for symbol path generation can also be added to
+         * `Highcharts.SVGRenderer.prototype.symbols`. The callback is then
+         * used by its method name, as shown in the demo.
+         *
+         * @sample {highcharts} highcharts/plotoptions/series-marker-symbol/
+         *         Predefined, graphic and custom markers
+         * @sample {highstock} highcharts/plotoptions/series-marker-symbol/
+         *         Predefined, graphic and custom markers
+         *
+         * @type      {string}
+         */
+        symbol: 'cluster',
+        /**
+         * The radius of the cluster marker.
+         *
+         * @sample {highcharts} highcharts/plotoptions/series-marker-radius/
+         *         Bigger markers
+         *
+         * @type      {number}
+         */
+        radius: 15,
```

Comment:
`@type` and `@default` are not needed, if it can be reflected from the default value, that follows.